### PR TITLE
[OSPP2024]feat: support aliyun OSS Sink Connector

### PR DESCRIPTION
**Pull request by limbo-24**

### CODE REVIEW

File: connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            String joinedString = values.stream().collect(Collectors.joining("\n"));
+            String absolutePath = key + objectName;
+            boolean exists = ossClient.doesObjectExist(bucketName, absolutePath);
+            long offset = 0;
+            // If the object does not exist, create it and set offset to 0, otherwise read the offset of the current object
+            if (exists) {
+                try {
+                    OSSObject ossObject = ossClient.getObject(bucketName, absolutePath);
+                    InputStream inputStream = ossObject.getObjectContent();
+                    offset = inputStream.available();
```
Comment (ingdex) [inline connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java:93]: Roll the file once its size exceeds the threshold. For example, when the file size becomes larger than 200 MB, create a new file for writing and archive the old ones.

File: connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    }
+
+    @Override
+    public void put(List<ConnectRecord> sinkRecords) throws ConnectException {
+        try {
+            sinkRecords.forEach(sinkRecord -> {
+                try {
+                    handleRecord(sinkRecord);
+                } catch (OSSException oe) {
+                    System.out.println("Caught an OSSException, which means your request made it to OSS, "
```
Comment (ingdex) [inline connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java:197]: Don't write msg to STDOUT, use a log instead.

File: connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+                } catch (ClientException ce) {
+                    System.out.println("Caught an ClientException, which means the client encountered "
+                            + "a serious internal problem while trying to communicate with OSS, "
+                            + "such as not being able to access the network.");
+                    System.out.println("Error Message:" + ce.getMessage());
+                } catch (Exception e) {
+                    log.error("OSSSinkTask | genObjectOffset | error => ", e);
+                }
+            });
+            if (enableBatchPut && !recordMap.isEmpty()) {
```
Comment (ingdex) [inline connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java:212]: When any exception happens above, I think processMap() should not be called.

File: connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java (hunk 1 above)
Comment (limbo-24) [reply connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java:97]: Before writing to OSS, check the offset size. If it exceeds OBJECT_SIZE_THRESHOLD, append the current timestamp as a suffix to the specified file name. For example: `test.txt -> test.txt_2024-09-21-22:33:00`

File: connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java (hunk 2 above)
Comment (limbo-24) [reply connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java]: Fixed: replaced `System.out.println` with `log.error`.

File: connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java (hunk 3 above)
Comment (limbo-24) [reply connectors/aliyun/rocketmq-connect-oss/src/main/java/org/apache/rocketmq/connect/oss/sink/OssSinkTask.java]: Added a boolean variable `hasException` to ensure `processMap` is called only if all records are processed successfully.

**Review by ingdex — APPROVED** [review]
LGTM

File: connectors/aliyun/rocketmq-connect-oss/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+|-----------------|--------|----------------|--------------------------------|----------------------------|
+| accountEndpoint | String | YES            | OSS endpoint                   | oss-cn-beijing.aliyuncs.com |
+| accessKeyId     | String | YES            | 阿里云授信账号的AK                     | xxxx                       |
+| accessKeySecret | String | YES            | 阿里云授信账号的SK                     | xxx                        |
+| bucketName      | String | YES            | OSS bucketName                 | test_bucket                |
+| objectName      | String | YES            | 上传目的object名字                   | test.txt                   |
+| region          | String | YES            | OSS region                     | cn-beijing                 |
+| partitionMethod | String | YES            | 分区模式，Normal表示不分区，Time表示按时间分区   | Time                       |
+| fileUrlPrefix   | String | YES            | 到object的URL前缀                  | file1/                     |
+| enableBatchPut  | String | NO             | 是否开启批处理模式                      | true                       |
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-oss/README.md:49]: 什么场景下enableBatchPut会设置为false？如果不存在这样的场景，可以移除这个参数。

Comment (limbo-24) [reply connectors/aliyun/rocketmq-connect-oss/README.md]: removed enableBatchPut

File: connectors/aliyun/rocketmq-connect-oss/pom.xml [hunk 5]
```diff
@@ -0,0 +1,218 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.rocketmq</groupId>
+    <artifactId>rocketmq-connect-oss</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (ingdex) [inline connectors/aliyun/rocketmq-connect-oss/pom.xml:9]: 使用release版本号

Comment (limbo-24) [reply connectors/aliyun/rocketmq-connect-oss/pom.xml]: fixed

**Comment by RockteMQ-AI** [comment]
**Issue Evaluation**

Category: `enhancement` | Status: **Needs Info**

This is a feature request for an Aliyun OSS Sink Connector (tagged OSPP2024). The issue body is empty, so we need more context to evaluate feasibility:
- [ ] Scope: What data formats should be supported (JSON, Avro, etc.)?
- [ ] Configuration: Expected sink properties (bucket, endpoint, auth)
- [ ] Delivery guarantees: At-least-once vs exactly-once
- [ ] Is there an existing PR or branch with initial work?
- [ ] OSPP2024 mentor and milestone timeline

Please provide these details so we can assess the implementation plan.

---
*Automated evaluation by @RockteMQ-AI*

**Pull request merged**